macOS/DuckDuckGo/Updates/UpdateController.swift

```diff
@@ -155,6 +186,20 @@ final class UpdateController: NSObject, UpdateControllerProtocol {
 #else
         checkForUpdateRespectingRollout()
 #endif
+
+        subscribeToResignKeyNotifications()
+    }
+
+    private var cancellables = Set<AnyCancellable>()
+
+    private func subscribeToResignKeyNotifications() {
+        NotificationCenter.default.publisher(for: NSWindow.didResignKeyNotification)
+            .debounce(for: .seconds(1), scheduler: DispatchQueue.main)
+            .sink { [weak self] _ in
+                self?.discardCurrentUpdateIfExpiredAndCheckAgain()
```

Comment:
If DuckDuckGo.app binary is owned by another user, this may lead to unexpected permission prompt and repeated download attempt if the user cancels it (see test case 4 [here](https://app.asana.com/1/137249556945/project/1202500774821704/task/1209475513300639?focus=true)).

![Screenshot 2025-05-20 at 1 19 57 AM](https://github.com/user-attachments/assets/7b2a096d-aedd-4cad-b573-33d5c5e7c95a)

Reply:
I can't find any difference between the test build and our release build.  I don't think this is an issue introduced by my changes.

Could you confirm?

Reply:
As discussed on the Zoom call, let’s make a task for this behavior, but by itself it’s not a blocker.